build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/JdkInfo.cs

```diff
@@ -40,6 +44,12 @@ public override bool Execute ()
 
 				Log.LogMessage (MessageImportance.Low, $"  {nameof (AndroidSdk.JavaSdkPath)}: {javaSdkPath}");
 
+				if (File.Exists (ConfigurationOperatingSystemProps.ItemSpec)) {
```

Comment:
This is one spot I did something strange: 
1. `Configuration.OperatingSystem.props` has to _exist_ for xa-prep-tasks to build
1. A custom MSBuild task in xa-prep-tasks needs to replace `@JAVA_HOME@` in `Configuration.OperatingSystem.props`, I put this in `JdkInfo` because it was doing the same thing for Java.Interop

Maybe there is a better way?

Reply:
We already have a task which handles `@...@` replacement: [`<ReplaceFileContents/>`](https://github.com/xamarin/xamarin-android/tree/master/build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/ReplaceFileContents.cs). I would suggest reusing that:

```xml
<ReplaceFileContents
    SourceFile="Properties\AssemblyInfo.cs.in"
    DestinationFile="$(IntermediateOutputPath)AssemblyInfo.cs"
    Replacements="@JAVA_HOME@=$(_TheJavaSdkPath)"
/>
```

This simply raises another question: where does `$(_TheJavaSdkPath)` come from? You should make `JdkInfo.JavaSdkPath` an `[Output]` property, which would allow you to set `$(_TheJavaSdkPath)` and use it in a later `<ReplaceFileContents/>` call.

Reply:
> 1. `Configuration.OperatingSystem.props` has to exist for xa-prep-tasks to build

What we could do is add another conditional import ("Indirections, all the way down!") to `Configuration.OperatingSystem.props`:

```xml
<Import
    Project="$(MSBuildThisFileDirectory)Configuration.Jdk.Props"
    Condition="Exists('$(MSBuildThisFileDirectory)Configuration.Jdk.Props')"
/>
```

You could then place the `$(JavaSdkDirectory)`, `$(JavaCPath)`, and `$(JarPath)` values within `build-tools\scripts\windows-Configuration.Jdk.Props.in`.